packages/react-components/react-tree/src/components/Tree/Tree.types.ts

```diff
@@ -108,10 +112,14 @@ export type TreeProps = ComponentProps<TreeSlots> & {
    *
    * FIXME: This method is not ideal, as navigation should be handled internally by tabster.
    *
+   * If a boolean is returned, it will be used to indicate if the navigation should be allowed.
+   * If false, the navigation will be prevented.
+   * If true or undefined, the navigation will be allowed.
+   *
    * @param event - a React's Synthetic event
    * @param data - A data object with relevant information,
    */
-  onNavigation?(event: TreeNavigationEvent_unstable, data: TreeNavigationData_unstable): void;
+  onNavigation?(event: TreeNavigationEvent_unstable, data: TreeNavigationData_unstable): void | boolean;
```

Comment:
do we have a concrete use case where we want to disable navigation?

Reply:
In that case is there a specific use case that `preventDefault` breaks to need to return a boolean?

I'm totally fine with returning a boolean value to prevent this if calling `preventDefault` breaks some other use case (i.e. HTML inputs)

Reply:
The problem with default prevention is that it'll already be default prevented to avoid scrolling on navigation. So there's no way to know if `preventDefault` means "don't scroll" or "do default behaviour but don't scroll" or "don't do default behaviour and don't scroll".

Reply:
This also goes for the sake of saying we probably need an universal way of opting-out of internal mechanisms that works for every v9 component.

Preventing default might not work for some cases (like Escape key pressing on modals with menu inside, right now we're stopping the propagation of those events to avoid colisions), and returning a boolean may also not work on those cases, we might need something more complex?!

Reply:
I'll convert this back into a draft as I believe this mechanism should be better discussed

Reply:
I'll keep this being done by default prevention for now as a conclusion for this 👍🏻